packages/slice-machine/server/src/api/services/sliceService.ts

```diff
@@ -0,0 +1,71 @@
+import path from "path";
+import type Models from "@slicemachine/core/build/src/models";
+import { resolvePathsToScreenshot } from "@slicemachine/core/build/src/libraries/screenshot";
+
+import { onError } from "../common/error";
+import { upload } from "./uploadScreenshotClient";
+import { BackendEnvironment } from "@lib/models/common/Environment";
+import DefaultClient from "@models/common/http/DefaultClient";
+import FakeClient from "@models/common/http/FakeClient";
+import { snakelize } from "@lib/utils/str";
+
+export const createOrUpdate = async (
+  slices: ReadonlyArray<Models.SliceAsObject>,
+  sliceName: string,
+  model: Models.SliceAsObject,
+  client: DefaultClient | FakeClient
+) => {
+  if (slices.find((e) => e.id === snakelize(sliceName))) {
+    return await client.updateSlice(model);
+  } else {
+    return await client.insertSlice(model);
+  }
+};
+
+export async function uploadScreenshots(
+  env: BackendEnvironment,
+  sliceModel: Models.SliceAsObject,
+  sliceName: string,
+  from: string
+): Promise<{ [variationId: string]: string | null }> {
+  const variationIds: string[] = sliceModel.variations.map(
+    (v: Models.VariationAsObject) => v.id
+  );
+
+  const imageUrlsByVariation: { [variationId: string]: string | null } = {};
+
+  for (let i = 0; i < variationIds.length; i += 1) {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
```

Comment:
how is this unsafe?

Reply:
it was not good catch, i think the unsafe warning came from the core not being rebuild